index.js

```diff
@@ -102,23 +102,27 @@ Client.prototype._write = function (obj, enc, cb) {
       this._chopper.chop(cb)
     }
   } else {
+    if ('transaction' in obj) {
+      truncate.transaction(obj, this._opts)
+    } else if ('span' in obj) {
+      truncate.span(obj, this._opts)
+    } else if ('error' in obj) {
+      truncate.error(obj, this._opts)
+    }
```

Comment:
Should these not point to the sub-object, not the base object? ie: `obj.transaction`?

Also, is there a reason to else-if these? Are we unable to send different types at the same time?

Reply:
I can see a case for sending a span and error at the same time. You wrap an async call in a span from call to callback, and an error occurs, so you report the error and span when the callback is reached. 🤔

Reply:
Version 2 of the APM Server intake API currently doesn't support that an object contains more than a single root property - either `transaction`, `span` or `error`.

Reply:
Ah, that's fine then. Might be worth considering for @elastic/apm-server to support in the future though.